modules/host/negotiaterevisecontract_test.go

```diff
@@ -0,0 +1,88 @@
+package host
+
+import (
+	"testing"
+)
+
+func TestShannonEntropy(t *testing.T) {
+	t.Parallel()
+	uniform := make([]byte, 256)
+	for i := 0; i < 256; i++ {
+		uniform[i] = byte(i)
+	}
+	almost := make([]byte, 256)
+	for i := 0; i < 256-10; i++ {
+		almost[i] = byte(i)
+	}
+	testCases := []struct {
+		name      string
+		input     []byte
+		low, high float64
```

Comment:
might be better to use `exp, delta float64` and then test with:
```go
if math.Abs(got - ts.exp) > ts.delta {
    // fail
}
```

Reply:
Done. Also delta is the same for all the test cases.